src/renderer/component/page/view.jsx

```diff
@@ -1,33 +1,96 @@
 // @flow
 import * as React from 'react';
 import classnames from 'classnames';
+import Spinner from 'component/common/spinner';
+import { isShowingChildren } from 'util/dom';
+
+// time in ms to wait to show loading spinner
+const LOADER_TIMEOUT = 1500;
 
 type Props = {
-  children: React.Node,
+  children: React.Node | Array<React.Node>,
   pageTitle: ?string,
   noPadding: ?boolean,
   extraPadding: ?boolean,
   notContained: ?boolean, // No max-width, but keep the padding
+  loading: ?boolean,
 };
 
-const Page = (props: Props) => {
-  const { pageTitle, children, noPadding, extraPadding, notContained } = props;
-  return (
-    <main
-      className={classnames('main', {
-        'main--contained': !notContained && !noPadding && !extraPadding,
-        'main--no-padding': noPadding,
-        'main--extra-padding': extraPadding,
-      })}
-    >
-      {pageTitle && (
-        <div className="page__header">
-          {pageTitle && <h1 className="page__title">{pageTitle}</h1>}
-        </div>
-      )}
-      {children}
-    </main>
-  );
+type State = {
+  showLoader: ?boolean,
 };
 
+class Page extends React.PureComponent<Props, State> {
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
```

Comment:
React recommends `getDerivedStateFromProps` moving forward instead of `componentWillReceiveProps`